src/xmipp/libraries/data/monogenic.cpp

```diff
@@ -0,0 +1,712 @@
+					{
+						DIRECT_MULTIDIM_ELEM(pMask, n) = 0;
+						DIRECT_MULTIDIM_ELEM(plocalResolutionMap, n) = resolution_2;
+					}
+				}
+		}
+	}
+
+// SETLOCALRESOLUTIONHALFMAPS: Set the local resolution of a voxel, by
+// determining if the monogenic amplitude "amplitudeMS" is higher than
+// the threshold of noise "thresholdNoise". Thus the local resolution
+// map "plocalResolution" is set, with the resolution value "resolution"
+// or with "resolution_2". "resolution" is the resolution for with
+// the hypohtesis test is carried out, and "resolution_2" is the resolution
+// of the analisys two loops ago (see monores method)
+void Monogenic::setLocalResolutionMap(const MultidimArray<double> &amplitudeMS,
+		MultidimArray<int> &pMask, MultidimArray<double> &plocalResolutionMap,
+		double &thresholdNoise, double &resolution, double &resolution_2)
+	{
+		FOR_ALL_DIRECT_ELEMENTS_IN_MULTIDIMARRAY(amplitudeMS)
+		{
+			if (DIRECT_MULTIDIM_ELEM(pMask, n)>=1)
+				if (DIRECT_MULTIDIM_ELEM(amplitudeMS, n)>thresholdNoise)
+				{
+					DIRECT_MULTIDIM_ELEM(pMask, n) = 1;
+					DIRECT_MULTIDIM_ELEM(plocalResolutionMap, n) = resolution;//sampling/freq;
+				}
+				else{
+					DIRECT_MULTIDIM_ELEM(pMask, n) += 1;
+					if (DIRECT_MULTIDIM_ELEM(pMask, n) >2)
+					{
+						DIRECT_MULTIDIM_ELEM(pMask, n) = -1;
+						DIRECT_MULTIDIM_ELEM(plocalResolutionMap, n) = resolution_2;//maxRes - counter*R_;
+					}
+				}
+		}
+	}
+
+
+// MONOGENICAMPLITUDE_3D_FOURIER: Given the fourier transform of a map
+// "myfftV", this function computes the monogenic amplitude "amplitude" 
+// iu is the inverse of the frequency in Fourier space.
+void Monogenic::monogenicAmplitude_3D_Fourier(const MultidimArray< std::complex<double> > &myfftV,
+		MultidimArray<double> iu, MultidimArray<double> &amplitude, int numberOfThreads)
+{
+	Matrix1D<double> freq_fourier_z, freq_fourier_y, freq_fourier_x;
+
+	iu = fourierFreqs_3D(myfftV, amplitude, freq_fourier_x, freq_fourier_y, freq_fourier_z);
+
+	// Filter the input volume and add it to amplitude
+	MultidimArray< std::complex<double> > fftVRiesz, fftVRiesz_aux;
+	fftVRiesz.initZeros(myfftV);
+	fftVRiesz_aux.initZeros(myfftV);
+	std::complex<double> J(0,1);
+
+	long n=0;
+	FOR_ALL_DIRECT_ELEMENTS_IN_MULTIDIMARRAY(myfftV)
+	{
+		//double H=0.5*(1+cos((un-w1)*ideltal));
+		DIRECT_MULTIDIM_ELEM(fftVRiesz, n) = DIRECT_MULTIDIM_ELEM(myfftV, n);
+		DIRECT_MULTIDIM_ELEM(fftVRiesz_aux, n) = -J;
+		DIRECT_MULTIDIM_ELEM(fftVRiesz_aux, n) *= DIRECT_MULTIDIM_ELEM(fftVRiesz, n);
+		DIRECT_MULTIDIM_ELEM(fftVRiesz_aux, n) *= DIRECT_MULTIDIM_ELEM(iu, n);
+
+	}
+	MultidimArray<double> VRiesz;
+	VRiesz.resizeNoCopy(amplitude);
+
+	FourierTransformer transformer_inv;
+	transformer_inv.inverseFourierTransform(fftVRiesz, VRiesz);
+
+	FOR_ALL_DIRECT_ELEMENTS_IN_MULTIDIMARRAY(amplitude)
+		DIRECT_MULTIDIM_ELEM(amplitude,n) = DIRECT_MULTIDIM_ELEM(VRiesz,n)*DIRECT_MULTIDIM_ELEM(VRiesz,n);
+
+	// Calculate first component of Riesz vector
```

Comment:
this looks quite similar to `amplitudeMonoSig3D_LPF`

Reply:
it is very similar, one of them implements a filter, and the other one is faster. They are two different function because  the fast one is called many times. Otherwise, they would be defined as the same function